src/metrics/rules/active_ruleset.go

```diff
@@ -144,46 +135,72 @@ func (as *activeRuleSet) ForwardMatch(
 func (as *activeRuleSet) forwardMatchAt(
 	id []byte,
 	timeNanos int64,
-	matchOpts MatchOptions,
-) (forwardMatchResult, error) {
-	mappingResults, err := as.mappingsForNonRollupID(id, timeNanos, matchOpts)
-	if err != nil {
-		return forwardMatchResult{}, err
+	matchOpts *MatchOptions,
+) error {
+	if err := as.mappingsForNonRollupID(id, timeNanos, matchOpts); err != nil {
+		return err
 	}
-	rollupResults, err := as.rollupResultsFor(id, timeNanos, matchOpts)
-	if err != nil {
-		return forwardMatchResult{}, err
+
+	if err := as.rollupResultsFor(id, timeNanos, matchOpts); err != nil {
+		return err
+	}
+
+	ensureNonDecreasingCutoverForExistingID(matchOpts.MatchResult, timeNanos)
+	dedupeForExistingIDStagedMetadatas(matchOpts.MatchResult)
```

Comment:
This is the `unique` found here: https://github.com/m3db/m3/pull/4019/files#diff-2d6d0ddef3f98e8ba7f8d8a15c0367b893bd32dec1fcddf062c3c9034eca966bL159